app/controllers/assets_controller.rb

```diff
@@ -111,14 +111,19 @@ def new
   end
 
   def edit
-    @descriptionless = @user.assets.descriptionless
+    @descriptionless = @user.assets.not_current(@asset.id).descriptionless if @user.assets.not_current(@asset.id).descriptionless.count > 1
     @allow_reupload = true
     render 'edit_white' if white_theme_enabled?
   end
 
   def mass_edit
     redirect_to_default && (return false) unless logged_in? && (current_user.id == @user.id) || admin?
-    @descriptionless = @user.assets.descriptionless
+    # currently we redirect asset # publish to mass_edit with params["assets"]
```

Comment:
@innatewonderer any chance you can elaborate on this comment? I'm wondering if the app is doing something it shouldn't be doing. Is it related to the logic below?

Reply:
When you click `publish` for a new track upload, I'd expect it to redirect to `/assets/edit` but instead you get redirected to `/mass_edit`... Wasn't sure if it's a feature or a bug. Wasn't sure if there is a way to upload multiple tracks at once or a desire to have such ability :)

Reply:
Ah good call. Yeah, one track should go to /edit, multiple should go to /mass_edit — right now you can upload a zip file — and actually if we change the file selector to a multi on the front end, you can select multiple there too. Drag and drop coming Jan 2019 care of Manfred.

Reply:
I'm going to merge this for now, but ideally we would clean this up ASAP, as it makes logic in the controller action more difficult to understand. Maybe the "descriptionless" stuff can be extracted into a protected method or even tucked away into a model so that the controller can be more readable.